pycbc/io/live.py

```diff
@@ -345,10 +344,6 @@ def upload(self, fname, gracedb_server=None, testing=True,
         labels: list
             Optional list of labels to tag the new event with.
         """
-        import matplotlib
-        matplotlib.use('Agg')
```

Comment:
I am afraid this must remain here, otherwise the matplotlib calls made by `generate_snr_plot()` will not (necessarily) be made with the Agg backend.

Maybe the backend should rather be set in `pycbc_live` straight at initialization, rather than deep inside a module like this…

Reply:
It would be better in pycbc_live I think, just in case the module is being called elsewhere

Reply:
(which is the case I think?)

Reply:
Ah, `pycbc_live` does already set the Agg backend in fact. However, `pycbc_optimize_snr` does not.

Reply:
I am opening a separate PR to address this, as it is more a PyCBC Live thing than offline.

Reply:
I will rebase once that is in then